lib/use-checkout-pricing.js

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from 'react';
+import useRecurly from './use-recurly';
+
+/**
+ * useCheckoutPricing interface
+ *
+ * Defines
+ *  1. `initialInputs` argument in `useCheckoutPricing`
+ *  2. `newInputs` argument in `updateInputs`
+ *  3. `useEffect` dependencies (not yet added)
+ *
+ * @typedef {Object} useCheckoutPricingInterface
+ * @property {Array} subscriptions
+ * @property {Array} adjustments
+ */
+
+/**
+ * A custom hook for interfacing with react-recurly's checkoutPricing API meant
+ * to mimic the return values and behavior of `react.useState`.
+ *
+ * Accepts an `initialInputs` param (same as useState).
+ *
+ * Returns a tuple with a pricing object, a setter for updating pricing inputs
+ * (same as useState), and a loading state represented as a boolean.
+ *
+ * @typedef {Object} pricing
+ * @typedef {Function} setPricing
+ * @typedef {Boolean} loadingState
+ * @typedef {[pricing, setPricing, loadingState]} useCheckoutPricingInstance
+ *
+ * @param {PricingInput} useCheckoutPricingInterface
+ * @returns {useCheckoutPricingInstance} useCheckoutPricingInstance
+ */
+export default function useCheckoutPricing(initialInputs = {}, handleError = throwError) {
+  const recurly = useRecurly();
+  const [pricing, setPricing] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [inputs, setInputs] = useState(initialInputs);
+
+  useEffect(() => {
+    setLoading(true);
+    const { subscriptions = [], adjustments = [] } = inputs;
+    let checkoutPricing = recurly.Pricing.Checkout();
+
+    if (adjustments.length) {
+      checkoutPricing = addAdjustments(adjustments, checkoutPricing);
+    }
+
+    addSubscriptions(subscriptions, checkoutPricing)
+      .then(() => {
+        setPricing(checkoutPricing);
+      })
+      .catch(handleError)
+      .finally(() => setLoading(false));
+  }, [inputs]);
+
+  function addAdjustments(adjustments, checkoutPricing) {
+    return adjustments
+      .reduce((checkoutPricing, adjustment) => {
+        return checkoutPricing.adjustment(adjustment);
+      }, checkoutPricing)
+      .catch(handleError);
+  }
+
+  function addSubscriptions(subscriptions, checkoutPricing) {
+    const { subscriptionPricings } = subscriptions.reduce(
+      ({ checkoutPricing, subscriptionPricings }, { plan, taxCode, addons = [], quantity }) => {
+        let subscriptionPricing = recurly.Pricing.Subscription().plan(plan, { quantity });
+
+        if (addons.length) {
+          subscriptionPricing = addAddons(addons, subscriptionPricing);
+        }
+
+        if (taxCode) {
+          subscriptionPricing = subscriptionPricing.tax(taxCode);
```

Comment:
The `tax` method here needs some refinement to comply with its `signature`.

Perhaps we just allow a mixed object to be passed in directly, which itself should be in the form `{ taxCode, vatNumber }`, but the exact shape of that signature can be validated by `SubscriptionPricing`